src/lamp_py/ingestion/glides.py

```diff
@@ -0,0 +1,405 @@
+        """Add incoming records to a local parquet file"""
+        process_logger = ProcessLogger(
+            process_name="append_glides_records", type=self.type
+        )
+        process_logger.log_start()
+
+        new_dataset = self.convert_records()
+
+        joined_ds = new_dataset
+        if os.path.exists(self.local_path):
+            joined_ds = pd.dataset([pd.dataset(self.local_path), new_dataset])
+
+        process_logger.add_metadata(
+            new_records=new_dataset.count_rows(),
+            total_records=joined_ds.count_rows(),
+        )
+
+        now = datetime.now(tz=timezone.utc)
+        start = datetime(2024, 1, 1, tzinfo=timezone.utc)
+
+        new_path = os.path.join(self.tmp_dir, f"new_{self.base_filename}")
+        row_group_count = 0
+        with pq.ParquetWriter(new_path, schema=joined_ds.schema) as writer:
+            while start < now:
+                end = start + relativedelta(months=1)
+                if end < now:
+                    table = joined_ds.filter(
+                        (pc.field("time") >= start) & (pc.field("time") < end)
+                    ).to_table()
+                else:
+                    table = joined_ds.filter(
+                        (pc.field("time") >= start)
+                    ).to_table()
+
+                if table.num_rows > 0:
+                    row_group_count += 1
+                    writer.write_table(table)
+
+                start = end
+
+        process_logger.add_metadata(row_group_count=row_group_count)
+
+        os.replace(new_path, self.local_path)
+
+        upload_file(file_name=self.local_path, object_path=self.remote_path)
+
+        process_logger.log_complete()
+
+
+class EditorChanges(GlidesConverter):
+    """
+    Converter for Editor Change Events
+    https://mbta.github.io/schemas/events/glides/com.mbta.ctd.glides.editors_changed.v1
+    """
+
+    def __init__(self) -> None:
+        GlidesConverter.__init__(self, base_filename="editor_changes.parquet")
+
+    @property
+    def event_schema(self) -> pyarrow.schema:
+        """
+        Editor Changes Schema is the base Kinesis data plus metadata plus a
+        list of changes objects.
+        """
+        glides_editor_change = pyarrow.struct(
+            [
+                ("type", pyarrow.string()),
+                ("location", self.glides_location),
+                ("editor", self.glides_user),
+            ]
+        )
+
+        return pyarrow.schema(
+            [
+                (
+                    "data",
+                    pyarrow.struct(
+                        [
+                            ("metadata", self.glides_metadata),
+                            ("changes", pyarrow.list_(glides_editor_change)),
+                        ]
+                    ),
+                ),
+                ("id", pyarrow.string()),
+                ("type", pyarrow.string()),
+                ("time", pyarrow.timestamp("ms", tz="UTC")),
```

Comment:
should drop the `tz` from these fields, based on our parquet -> tableau experience

Reply:
good call. i'll do a conversion here.

Reply:
i built out the localization logic, but then realized we probably want to do that in a downstream application rather than in the ingestion app here.